src/Service/CategoriesService.php

```diff
@@ -18,39 +18,79 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
-    public function findFormats(): array
+    public function findFormats($ttl = CacheInterface::NORMAL): array
     {
-        $formats = $this->repository->findAllByTypeAndMaxDepth('format', 2);
-        return $this->mapManyEntities($formats);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () {
+                $formats = $this->repository->findAllByTypeAndMaxDepth('format', 2);
+                return $this->mapManyEntities($formats);
+            }
+        );
     }
 
-    public function findGenres(): array
+    public function findGenres($ttl = CacheInterface::NORMAL): array
     {
-        $genres = $this->repository->findAllByTypeAndMaxDepth('genre', 2);
-        return $this->mapManyEntities($genres);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () {
+                $genres = $this->repository->findAllByTypeAndMaxDepth('genre', 2);
+                return $this->mapManyEntities($genres);
+            }
+        );
     }
 
-    public function findFormatByUrlKeyAncestry(string $formatUrlKey): ?Format
+    public function findFormatByUrlKeyAncestry(string $formatUrlKey, $ttl = CacheInterface::NORMAL): ?Format
     {
-        $format = $this->repository->findByUrlKeyAncestryAndType([$formatUrlKey], 'format');
-        return $this->mapSingleEntity($format);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $formatUrlKey, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($formatUrlKey) {
+                $format = $this->repository->findByUrlKeyAncestryAndType([$formatUrlKey], 'format');
+                return $this->mapSingleEntity($format);
+            }
+        );
     }
 
-    public function findGenreByUrlKeyAncestry(array $urlHierarchy): ?Genre
+    public function findGenreByUrlKeyAncestry(array $urlHierarchy, $ttl = CacheInterface::NORMAL): ?Genre
     {
-        $genre = $this->repository->findByUrlKeyAncestryAndType($urlHierarchy, 'genre');
-        return $this->mapSingleEntity($genre);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, implode($urlHierarchy), $ttl);
```

Comment:
It was my intention that each part of the key was split by a hyphen. If I imploded the urlHierarchy by hyphen too, it would not make the key as clear if we were to inspect it.